src/faultlog/faultlog_main.cpp

```diff
@@ -0,0 +1,59 @@
+#include <faultlog_policy.hpp>
+#include <fco_records.hpp>
+#include <guarded_without_eid.hpp>
+#include <libguard/guard_interface.hpp>
+#include <libguard/include/guard_record.hpp>
+#include <nlohmann/json.hpp>
+#include <sdbusplus/bus.hpp>
+#include <servicable_records.hpp>
+#include <unresolved_pels.hpp>
+
+#include <iostream>
+
+using namespace openpower::faultlog;
+using ::nlohmann::json;
+using ::openpower::guard::GuardRecords;
+
+#define GUARD_RESOLVED 0xFFFFFFFF
+
+int main(int /*arg*/, char** /*argv*/)
+{
+    try
+    {
+        auto bus = sdbusplus::bus::new_default();
+
+        nlohmann::json faultLogJson = json::array();
+
+        openpower::guard::libguard_init();
+        // Don't get ephemeral records because those type records are not
+        // intended to expose to the end user, just created for internal purpose
+        // to use by the BMC and Hostboot.
+        openpower::guard::GuardRecords records = openpower::guard::getAll(true);
+        GuardRecords unresolvedRecords;
+        // filter out all unused or resolved records
+        for (const auto& elem : records)
+        {
+            if (elem.recordId != GUARD_RESOLVED)
+            {
+                unresolvedRecords.emplace_back(elem);
+            }
+        }
+
+        std::vector<uint32_t> processedEId;
+        ServicableRecords::populate(bus, unresolvedRecords, processedEId,
+                                    faultLogJson);
+        GuardedWithoutEidRecords::populate(unresolvedRecords, faultLogJson);
+        FCORecords::populate(faultLogJson);
+        FaultLogPolicy::populate(bus, faultLogJson);
+        UnresolvedPELs::populate(bus, unresolvedRecords, processedEId,
+                                 faultLogJson);
+        std::cout << "Nag data is " << std::endl;
```

Comment:
Would it remain as cout?? Can we switch it to log<>?

Reply:
we will write to std out from there debug-collector will pick and write a dump file

Reply:
https://github.com/openbmc/phosphor-debug-collector/blob/master/tools/dreport.d/plugins.d/diskusage

something like this